src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java

```diff
@@ -225,18 +224,104 @@ public PaymentMethodsApi(final User user) {
+            } else {
+                final PaymentMethod active = paymentMethod.activate();
+                LOG.debug("PaymentMethod successfully activated!");
+                json = new JsonPaymentMethod(active, Boolean.FALSE);
+            }
+            response = ResponseEntity.ok(json.toString());
+        } catch (final IllegalStateException exception) {
+            final String message = exception.getMessage();
+            if (message != null) {
+                response = ResponseEntity.badRequest().body(message);
+            } else {
+                response = ResponseEntity.badRequest().build();
+            }
+        }
+        return response;
+    }
+
+    /**
+     * This will deactivate the specified Stripe PaymentMethod.
+     *
+     * @param owner Owner of the project (login of user or org name).
+     * @param name Repo name.
+     * @param paymentMethodId Id of the PaymentMethod to be deactivated.
+     * @return Deactivated PaymentMethod in JSON.
+     */
+    @PutMapping(
+        value = "/projects/{owner}/{name}/wallets/stripe/paymentMethods/"
+            + "{paymentMethodId}/deactivate",
+        consumes = MediaType.APPLICATION_JSON_VALUE,
+        produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<String> deactivateStripePaymentMethod(
+        @PathVariable final String owner,
+        @PathVariable final String name,
+        @PathVariable final String paymentMethodId
+    ) {
+        LOG.debug(
+            "Deactivating Stripe PaymentMethod " + paymentMethodId
+                + " of Project " + owner + "/" + name + "... "
+        );
+        ResponseEntity<String> response;
+        try {
+            final PaymentMethod paymentMethod = this
+                .getStripePaymentMethod(owner, name, paymentMethodId)
+                .deactivate();
+            LOG.debug("PaymentMethod successfully deactivated!");
+            final JsonPaymentMethod json = new JsonPaymentMethod(
+                paymentMethod,
+                Boolean.FALSE
+            );
+            response = ResponseEntity.ok(json.toString());
+        } catch (final IllegalStateException exception) {
+            final String message = exception.getMessage();
+            if (message != null) {
+                response = ResponseEntity.badRequest().body(message);
+            } else {
+                response = ResponseEntity.badRequest().build();
+            }
+        }
+        return response;
+    }
+
+    /**
+     * Get a PaymentMethod from STRIPE wallet.
+     * @param owner Owner of the project (login of user or org name).
+     * @param name Repo name.
+     * @param paymentMethodId Id of the PaymentMethod to be deactivated.
+     * @return PaymentMethod.
+     * @throws IllegalStateException if method is not found.
+     * @checkstyle ExecutableStatementCount (60 lines).
+     */
+    private PaymentMethod getStripePaymentMethod(
```

Comment:
@amihaiemil Since checkings are the same for activate/deactivate I've extracted them into one method used by both endpoints. This also will ensure code coverage.